libs/user-identity-service/src/test/java/uk/gov/di/ipv/core/library/service/UserIdentityServiceTest.java

```diff
@@ -234,7 +251,28 @@ void checkNameCorrelationWithMissingNameCredentialsForReturnTrue() throws Except
     }
 
     @Test
-    void checkNameCorrelationWithSameBirthDatesAndMissingBirthDateCredentialsForReturnTrue()
+    void checkNameCorrelationWithMissingNameCredentialsForOnlyBAVCRIReturnFalse() throws Exception {
+        List<VcStoreItem> vcStoreItems =
+                List.of(
+                        createVcStoreItem(USER_ID_1, "ukPassport", SIGNED_VC_1, Instant.now()),
+                        createVcStoreItem(USER_ID_1, "dcmaw", SIGNED_VC_3, Instant.now()),
+                        createVcStoreItem(
+                                USER_ID_1,
+                                "bav",
+                                SIGNED_PASSPORT_VC_MISSING_BIRTH_DATE,
+                                Instant.now()));
+
+        when(userIdentityService.getVcStoreItems(USER_ID_1)).thenReturn(vcStoreItems);
+        mockCredentialIssuerConfig();
+
+        boolean isValid =
+                userIdentityService.checkNameAndFamilyNameCorrelationInCredentials(USER_ID_1);
+
+        assertFalse(isValid);
+    }
+
+    @Test
+    void checkBirthDateCorrelationWithSameBirthDatesAndMissingBirthDateCredentialsForReturnTrue()
```

Comment:
Similarly - I think this should return true?

I'm a bit scared by the fact there's an explicit test for this though, do you know if we're relying on this behaviour in a journey?

Reply:
Hi Joe, Thanks, you have brought up a very important point. There hasn't been a specific requirement from any of the other VC's to throw an error in the evidence when there is no name or birthdate. For name and birthdate, we gather and compare them for all VCs if they exist in the evidence, In summary, if name and birthdate not found in the evidence, the correlation check is not performed. I am sharing of the previous correlation epic and taks that we discussed. https://govukverify.atlassian.net/browse/PYIC-2600 https://docs.google.com/document/d/1LjET6Zw_JFwKNH2tODk8ZcZT2DLwMoWaEjeRIohXPWY/edit#heading=h.r15klbho5w4s  Additionally, some points were mentioned that we would address issues like pre-marital surname in the future. (https://gds.slack.com/archives/C0222KXB6RM/p1681832588075989) I have a concern. If we throw a correlation error or any other error when name and birthdate(not BAV) are not found in the evidence, it could impact a significant number of customer in production. Even when they enter different names in correlation at intervals, many people have asked us without understanding the error. I have concern about this. What do you think about getting approval from the Product team and Business teams and considering this in different task? Also, name correlation currently works on the same logic now.

Reply:
Yeah, we should make sure to confirm this change with product/business/architecture - do you want to do that, or shall I pursue?

The thing that reassures me is that SPOT have told me they _are_ currently checking that the name/birthdate exist on all VCs, and we should match their logic.

If we're really nervous, we could implement this as a two-step change:
- Add logic to check for missing name/birthdate, but log an _error_ in that case, rather than failing
- Deploy to production, and check how often we encounter that error
- If we don't see that error, then turn it into a proper failure - or handle the error properly

Reply:
Joe, I would greatly appreciate it if you could discuss and obtain approval from the relevant teams.. I am going to add  some checks and logs for names/birthdates.